### PR TITLE
Enhanced card details model

### PR DESCRIPTION
**Pull request by stpopa**
- Enhanced card details model with bank, cardCategory, cardCountry, cardFunding, cardScheme fields
- Using API version 5.6.0

### CODE REVIEW

File: Source/Model/JPCardDetails.h [hunk 1]
```diff
@@ -155,35 +155,60 @@ typedef NS_ENUM(NSUInteger, CardNetwork) {
...
+ */
+@property (nonatomic, strong) NSString *_Nullable cardCategory;
+
+/**
+ * Gets the country the card was issued from in ISO 3166-1 alpha-2 format (2 chararacters)
+ */
+@property (nonatomic, strong) NSString *_Nullable cardCountry;
+
+/**
+ * Gets or sets the funding type of the card.
```
Comment (zfoltin) [inline Source/Model/JPCardDetails.h]: Don't think we do much "setting" here as these come from the API, no? Probably just "Card funding" (maybe also mentioning some potential values (Debit / Credit)?)

File: Source/Model/JPCardDetails.h [hunk 2]
```diff
@@ -155,35 +155,60 @@ typedef NS_ENUM(NSUInteger, CardNetwork) {
...
+ */
+@property (nonatomic, strong) NSString *_Nullable cardFunding;
+
+/**
+ * Gets or sets the scheme of the card.
```
Comment (zfoltin) [inline Source/Model/JPCardDetails.h]: Probably just "The card scheme" (with some potential value, eg: Visa / MasterCard)

File: JudoKitObjCExampleApp/DetailViewController.m [hunk 3]
```diff
@@ -82,15 +82,40 @@ - (void)viewWillAppear:(BOOL)animated {
...
+        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Error"
+                                                                                 message:@"No card details data to display."
+                                                                          preferredStyle:UIAlertControllerStyleAlert];
+        [alertController addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleCancel handler:nil]];
+        [self presentViewController:alertController animated:YES completion:nil];
+        return;
+    }
+
+    NSArray<DetailsRow *> *rows = @[
+    [DetailsRow rowWithTitle:@"CardLastFour" value: cardDetails.cardLastFour],
```
[the remark this answers is not in the document]
Comment (stpopa) [reply JudoKitObjCExampleApp/DetailViewController.m]: > Probably worth making the title more human readable (eg: "Card last 4 digits", "End date", etc.)

why not to keep it consistent with the object properties?

File: JudoKitObjCExampleApp/DetailViewController.m [hunk 4]
```diff
@@ -82,15 +82,40 @@ - (void)viewWillAppear:(BOOL)animated {
...
+    [DetailsRow rowWithTitle:@"Bank" value: cardDetails.bank],
+    [DetailsRow rowWithTitle:@"CardCategory" value: cardDetails.cardCategory],
+    [DetailsRow rowWithTitle:@"CardCountry" value: cardDetails.cardCountry],
+    [DetailsRow rowWithTitle:@"CardFunding" value: cardDetails.cardFunding],
+    [DetailsRow rowWithTitle:@"CardScheme" value: cardDetails.cardScheme]
+    ];
+    
+    NSArray<DetailsSection *> *sections = @[[[DetailsSection alloc] initWithTitle:@"Card Details" rows: rows]];
+    DetailsTableViewController *detailsViewController = [[DetailsTableViewController alloc] initWithData:sections andTitle:@"Details"];
+    UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:detailsViewController];
```
[the remark this answers is not in the document]
Comment (stpopa) [reply JudoKitObjCExampleApp/DetailViewController.m]: > Does it need to be inside a navigation controller?

yes, I want a title there and a "Done" button

File: JudoKitObjCExampleApp/DetailsTableViewController.h [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+@interface DetailsSection: NSObject
+@property (nonatomic, strong) NSString *_Nullable title;
+@property (nonatomic, strong) NSArray<DetailsRow *> *rows;
+
+- (nonnull instancetype)initWithTitle:(nonnull NSString *)title
+                                 rows:(nonnull NSArray<DetailsRow *> *)rows;
+@end
+
+@interface DetailsTableViewController : UITableViewController
```
[the remark this answers is not in the document]
Comment (stpopa) [reply JudoKitObjCExampleApp/DetailsTableViewController.h]: hm, I'll check it

[the remark this answers is not in the document]
Comment (stpopa) [reply JudoKitObjCExampleApp/DetailsTableViewController.h]: seems like an issue with configurations in `.clang-format`, will sort it out and push it separate PR to not end up with a huge one here

**Pull request merged**